core-war/src/main/webapp/map/js/services/silverpeas-map-address-service.js

```diff
@@ -106,9 +106,15 @@
                 lonLat : lonLat,
                 street : osmResponse.address.road ? osmResponse.address.road.split(',') : [],
                 postalCode : osmResponse.address.postcode,
-                city : osmResponse.address.town,
+                city : osmResponse.address.city || osmResponse.address.town || osmResponse.address.village,
                 country : osmResponse.address.country,
               };
+              if (osmResponse.address.house_number) {
+                data.street[0] = osmResponse.address.house_number + ', ' + data.street[0];
+              }
```

Comment:
Be careful here: the format of the street depends on the localization; what your have written here is the format of a an address in France. In some other countries the house number comes after the street (for example in Germany). And in England, USA or Canada the number of the house and the street aren't separated by a coma but by a space. 

In my advice, we should use the l10n bundles to define a property for the format of a street and use it here.